mono/mini/aot-compiler.c

```diff
@@ -7375,6 +7376,39 @@ add_gsharedvt_wrappers (MonoAotCompile *acfg, MonoMethodSignature *sig, gboolean
 	}
 }
 
+static void mono_destroy_compile_aot (MonoAotCompile *acfg)
+{
+	GSList *curr = acfg->destroy_queue;
+
+	while (curr != NULL) {
+		mono_destroy_compile (curr->data);
```

Comment:
It would look better to iterate over the list, then delete it using g_slist_free ().